### DISCUSSION TITLE
Running the breakout board with both a spindle and diode laser

### DISCUSSION THREAD

**Discussion by axg20202** [phil-barrett/grblHAL-teensy-4.x#115] [Q&A]
Hi. I currently have a Teensy breakout board running my PrintNC CNC spindle over ModBus.  I want to try adding a laser module. As my spindle is controlled by ModBus to my VFD, can I avoid adding a relay to switch between the two 'spindles' or will they both be powered regardless of controller state? I understand that I need firmware compiled for the 2 spindles. and to change the mode of the controller to Laser mode, but is that all?

**Comment by phil-barrett** [comment]
There are a number of ways to control both.  Have you seen [this blog](https://www.grbl.org/single-post/using-a-diode-laser-with-grblhal)?  It uses a relay which I know you didn't want.  By the way there is an error in the LED indicator for spindle in the circuit (it will never go on). However, the relay itself has an LED that lights when the relay is energized (Spindle On).  Depending on the relay used, one might have to switch the NO and NC terminals. Alternatively, one could use a DPST switch and use on of the poles for turning on the spindle LED.

There are probably ways to switch between two spindles - take a look [here](https://github.com/grblHAL/Plugins_spindle/). It may give you a way to use a macro to switch spindles (macros can have M commands). (I am using spindle to indicate a device that is controlled via the Sxxxx command in GCode).

I like the idea of having a visual way to see what has been selected which is why I lean towards the relay approach.

**Comment by axg20202** [comment]
Thanks Phil. What I don't understand is that it sounds like a spindle needs to be activated in the controller to be used, but also all connected spindles are permanently electrically active anyway, hence the need for electrical switching? Seems odd. I had hoped that when the PMW 'spindle' is not the active spindle (as per gcode, iosender settings) it doesn't fire, and similarly the modbus spindle won't run if not the active spindle. The spindle plugin info doesn't really explain the relationship between software activation and electrical output. Or are these simply two different methods to switch?

**Reply by phil-barrett** [reply]
I think if you are using Modbus, it may not require relay switching.  I'm more worried about the laser getting switched on when using PWM or 0-10V (which uses PWM) for the actual spindle.  IIRC, when using Modbus spindle, PWM doesn't get output.

**Reply by axg20202** [reply]
Thanks Phil. Will give it a try. Glasses on, hope for the best!
